### DISCUSSION TITLE
Can I get more insights into the cloning process?

### DISCUSSION THREAD

**Discussion by oliversalzburg** [renovatebot/renovate#11665] [Request Help]
I'm trying to set up a new repository with renovate and iteration on the config is taking very long, because the initialization of the repository in the CI runs takes 10-15m.

```shell
DEBUG: Initializing git repository into /tmp/renovate/repos/gitlab/fairmanager/images (repository=fairmanager/images)
DEBUG: git clone completed (repository=fairmanager/images)
       "durationMs": 935717
```

I've only used renovate on one other repository before, which is a much larger (in commits, file count, storage size) project and it takes hardly any time:

```shell
DEBUG: Initializing git repository into /tmp/renovate/repos/gitlab/fairmanager/core-web3 (repository=fairmanager/core-web3)
DEBUG: git clone completed (repository=fairmanager/core-web3)
       "durationMs": 3901
```

I also don't remember the prior project ever taking this long to clone. So this is rather unexpected. It's so consistent, that I'm assuming something is weird about the new repo.

How can I troubleshoot this? When I clone the project on my machine it takes seconds.

**Comment by rarkins** [comment]
It performs a blobless clone. Make sure you are using the latest git version too

**Reply by oliversalzburg** [reply]
I realized yesterday that blobless clone is used after coming across https://github.com/renovatebot/renovate/pull/11401
It doesn't appear to matter though. I can clone the repo any way I want on my machine and it never takes more than a few seconds, while in CI it takes 15 minutes every time I run the job.

All CI runners use the same version of git. So I'm confused about the drastic difference in performance. Upgrading to a later version that's not in the Linux distributions repositories is a bit challenging. It doesn't appear promising without understanding why one repo would behave so differently.

**Reply by rarkins** [reply]
It could potentially be an edge case that's only triggered by that repo and with an older version of git. What version is being run right now? (it could potentially also be a git server-side bug and not git CLI)

You can do or not do whatever you decide, but in v27 we bumped the minimum git version to v2.33.0 and don't support earlier versions.

In the hosted app where we process 10s of thousands of repos, we average 3s clone per repo (github.com).

**Reply by oliversalzburg** [reply]
Very interesting. Thanks a lot for that.

Our GitLab instance has 2.25.1 available on the shell, but I'm not sure if GitLab somehow brings their own implementation. The renovate job uses the `renovate/renovate` image. The version I have has git 2.33.0.

On my machine, I use a 2.25.1 in WSL and a 2.33.0 on Windows.

As I don't see many options other than looking into upgrading at this point. I'll report back if it resolves the problem.

**Reply by viceice** [reply]
For ubuntu you can use https://launchpad.net/~git-core ppa to get latest git

https://launchpad.net/~git-core/+archive/ubuntu/ppa

**Reply by rarkins** [reply]
If you're using the `renovate/renovate` image then it should mean that there's no problem with the "client" version.

Can you replicate the clone from your own computer using e.g. the following approach?

```
docker run --rm -it renovate/renovate:slim /bin/bash
ubuntu@61233f19c0fe:/usr/src/app$ cd ~
ubuntu@61233f19c0fe:~$ git clone --filter=blob:none https://gitlab.com/renovate-bot/renovate-runner
Cloning into 'renovate-runner'...
warning: redirecting to https://gitlab.com/renovate-bot/renovate-runner.git/
remote: Enumerating objects: 2756, done.
remote: Counting objects: 100% (881/881), done.
remote: Compressing objects: 100% (68/68), done.
remote: Total 2756 (delta 869), reused 817 (delta 813), pack-reused 1875
Receiving objects: 100% (2756/2756), 269.17 KiB | 1.75 MiB/s, done.
Resolving deltas: 100% (1790/1790), done.
warning: redirecting to https://gitlab.com/renovate-bot/renovate-runner.git/
remote: Enumerating objects: 13, done.
remote: Counting objects: 100% (6/6), done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 13 (delta 0), reused 0 (delta 0), pack-reused 7
Receiving objects: 100% (13/13), 45.67 KiB | 4.57 MiB/s, done.
```

(but try it for the problematic repo)

**Reply by rarkins** [reply]
If you can replicate the slow clone then it would seem like a GitLab problem if they're using an outdated `git`. If you can't replicate the slow clone, then it remains a mystery as to what's going wrong.

**Reply by oliversalzburg** [reply]
Running the clone inside that container per your instructions also takes just seconds. Thanks for the support.

I'll probably not need to run the job at high frequencies anymore now, but I assume I will continue to run into problems. If I find out the cause, I'll report back.

**Reply by oliversalzburg** [reply]
After leaving the system alone for the weekend, the past 2 scheduled runs of the problematic repo only took seconds:
```
DEBUG: git clone completed (repository=fairmanager/images)
       "durationMs": 3924
```
I have no idea what (apparently temporary) condition caused the slow clones before.
